sbi/utils/user_input_checks.py

```diff
@@ -372,6 +372,23 @@ def check_prior_support(prior):
         )
 
 
+def check_embedding_net_device(embedding_net: nn.Module, batch_y: torch.Tensor) -> None:
+    batch_y_device = batch_y.device
```

Comment:
what if the user passes data on GPU? Then this check would pass and the embedding would remain on the GPU and cause issues with the density estimator that is constructed on CPU, no? 

If we assume that we always build the density estimator on the CPU, then we could just always move the embedding to the GPU, no? (and warn, or throw an error depending on whether we want to `warn and correct` or not. )

Reply:
I'm not sure this situation will arise in the sense that the mismatch between `prior` and `estimator` is checked before that step? So before this is ran we know that `batch_y` (generated from the `prior`) is the same device as `estimator`, and all that remains to do is to check that the `embedder` is also in that case? I guess the question is: is there  use case -unknown to me- where the `build` functions are used outside of the `classic` "pipeline" ?